--- 
title: "Phylogenetic Biology"
author: "Casey W. Dunn"
github-repo: caseywdunn/phylogenetic_biology
twitter-handle: caseywdunn
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
description: "An introduction to Phylogenetic Biology."
fig_caption: yes
always_allow_html: true
---

```{r preliminaries, include=FALSE}
	# The following should be installed from github with the specified 
	# devtools command. You will need to install devtools first.
	library( treeio ) # devtools::install_github("GuangchuangYu/treeio")
	library( ggtree ) # devtools::install_github("GuangchuangYu/ggtree")
	library( ape )
  library( geiger )
	library( gridExtra )
	library( magrittr )
	library( phytools )
	library( stringr )
	library( tidyverse )
  library( Matrix )
  library( phangorn )
  library( kableExtra )
  library( ggrepel )
  library( scales )

	source( "functions.R" )
	set.seed( 23456 )
```


```{r setup, include=FALSE}
knitr::opts_chunk$set(
  message = FALSE,
  warning = FALSE,
  echo = FALSE,
  dpi = 300,
  cache = FALSE
  )
```

```{r include=FALSE}

  theme_set(theme_classic())

  # nucleotide_palette = scales::hue_pal() 
  nucleotide_palette = scales::grey_pal()

  if ( knitr::is_latex_output() ){
    # PDF specific settings
    nucleotide_palette = scales::grey_pal()
    
  } else{
    # github specific settings
  }

```



```{r include=FALSE}
# automatically create a bib database for R packages
knitr::write_bib(c(
  .packages(), 'bookdown', 'knitr', 'rmarkdown', 'ape', 'geiger', 'ggtree', 'treeio', 'phytools'
), 'packages.bib')
```


# Preface {-}

## Approach

Models are at the core of modern phylogenetic biology. Models are usually taught in the context of phylogenetic inference, where they are used to look backward in time. This seems to me a bit like teaching students to master driving in reverse before you show them how to drive forward down a road. Phylogenetic models are generative - they explain how data are created under specific evolutionary processes. They are therefore more intuitive to understand in the context of generating data, proceeding in time from ancestral states forward to future states. I therefore focus first on building an intuitive understanding of models in the context of simulation, and only after that apply models to inference and other tasks that look back in time.

## Using this book

This book is intended both for the self directed learner, and for use in a course. 

I wrote it as a text for my course, Phylogenetic Biology (Yale EEB354). We read one chapter a week. We review and discuss the reading on Tuesdays, and then on Thursdays do hands-on work, read papers from the literature, or share student projects.

## Acknowledgements

Thanks in particular to the students of Yale EEB354 in the falls of 2020 (the first course I taught fully online) and 2022. This book started as a collection of lecture notes for this course. The students provided invaluable motivation, feedback, and patience. Thanks in particular to Lauren Mellenthin, the graduate teaching fellow for the course in 2020, and Namrata Ahuja, the teaching fellow in 2022. Members of my lab provided very helpful feedback when I posted new chapters. Steve Haddock and Felipe Zapata also provided close reads of most chapters, often within hours of completing first drafts. Thanks to Richard Hammack, author of Book of Proof, for his helpful advice on self publishing to facilitate student access.

## Dedication

This book is dedicated to my sister Jenny Dunn, who has been there every step of the way. I am so grateful for her adventurous spirit, quick mind, and shared love of life. Thank you Jenny.


## Source code

The source code for this manuscript is available on github at https://github.com/caseywdunn/phylogenetic_biology . The book was rendered from the source with `bookdown` [@bookdown2016].

Please submit any errors you find, typos, or suggestions that you have for improving the manuscript to the issue tracker at https://github.com/caseywdunn/phylogenetic_biology/issues .

## Other resources

The following sites have a wide variety of material that is relevant to the 
theory and and practice of phylogenetic biology.

- An extensive list of tools, tutorials, and examples of phylogenetic tools in 
the programming language R maintained by Briam O'Meara: https://cran.r-project.org/web/views/Phylogenetics.html

- The Workshop on Molecular Evolution at Woods Hole. This is an intensive summer 
course on phylogenetics, with an emphasis on building phylogenetic trees. Check 
out the faculty pages for lecture pdfs: https://molevolworkshop.github.io

- The Applied Phylogenetics Workshop in Bodega Bay. This is another summer course 
on phylogenetics, but with a bit more emphasis on using phylogenies to test 
evolutionary questions: http://treethinkers.org/tutorials/

The following books are great resources for learning more about phylogenetic biology:

- Baum, D. and Smith, S. (2012). Tree Thinking. An Introduction to Phylogenetic 
Biology. [Roberts Publishers](http://www.roberts-publishers.com/tree-thinking-an-introduction-to-phylogenetic-biology.html).

- Paradis, E. (2011) Analysis of Phylogenetics and Evolution with R. 
[Springer](http://www.springer.com/life+sciences/evolutionary+%26+developmental+biology/book/978-0-387-32914-7)

- Felsenstein, J. (2004) Inferring phylogenies. 
[Sinauer Associates](http://www.sinauer.com/detail.php?id=1775).

- Swofford, D. L., Olsen, G. J., Waddell, P. J., & Hillis, D. M. (1996). 
Phylogenetic inference. In: Molecular Systematics, Second Edition. eds: D. M. 
Hillis, C Moritz, & B. K. Mable. [Sinauer Associates](http://www.sinauer.com/detail.php?id=1775)

The following books provide general computational background for the topics covered here:

- Wickham, H., Grolemund, G (2017) R for Data Science. https://r4ds.had.co.nz

- Haddock, S. H. D. and Dunn, C. W. (2010). Practical Computing for Biologists. http://practicalcomputing.org





## License

![](figures/by-nc-nd-600.png)

This work is licensed under the [Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International License]( http://creativecommons.org/licenses/by-nc-nd/4.0/).
