---
title: "Exercise 04"
output: html_document
---

```{r setup, echo = FALSE, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ape)
```
## Introduction

The goals of this exercise are to:
- Run iqtree on the cluster
- Infer a phylogeny from an existing multiple sequence alignment with iqtree
- View and assess the phylogenetic tree and log file generated by iqtree

[iqtree](http://www.iqtree.org/) is a popular phylogenetic inference tool that searches for maximum likelihood trees. It takes in aligned DNA sequences, and outputs a phylogeny and information about the analysis. It is fast, well engineered, has a variety of very useful features, and has [excellent documentation](http://www.iqtree.org/doc/). 

This exercise builds on iqtree [minumum examples](http://www.iqtree.org/doc/Quickstart#minimal-command-line-examples). The [iqtree tutorial](http://www.iqtree.org/doc/Tutorial) provides good background for understanding an iqtree analysis.


## Getting set up

Please see last week's exercise for detail on how to log into the cluster and get around at the command line. Transfer the exercise files to a folder where you will analyze them. Open a command line and `cd` to the directory.

## Inspecting the files

There are two files. The first, `siph16s.phy`, contains the multiple sequence alignment. This is the input data that will be clamped at the tips of the tree. These sequences are for [siphonophores](http://www.siphonophores.org), and the [data were collected for my PhD thesis](http://dx.doi.org/10.1080/10635150500354837).

> **_TASK:_** Take a look at the alignment with the command `cat siph16s.phy`, which dumps the contents of the file to your window. 

These sequences are in [phylip format](https://evolution.genetics.washington.edu/phylip/doc/sequence.html). The first line has two numbers - the number of sequences (which will correspond to the number of tips in the phylogeny). The second is the number of sites in the sequences. The following lines start with the tip name, and then the sequence for the tip. In addition to nucleotides, the sequences also include `-`, which are gaps. These represent missing data or insertion/deletion events in the sequences.

> **_TASK:_** Now take a look at the job script with `cat job.sh` 

There are a few things that differ from the job file for last exercise. `iqtree` can use multiple CPUs to speed things up. So we specify `#SBATCH --cpus-per-task=8` at the top of the job script to allocate 8 CPUs to the job. We also load the iqtree module. And finally we call `iqtree`:

    iqtree2 -s siph16s.phy -nt AUTO

The `-s siph16s.phy` portion of this command specifies the input sequence alignment file name. `-nt AUTO` sets the number of threads to `AUTO`. This allows iqtree to automatically identify the optimum number of the 8 CPUs we allocated to use for the job.


## Run the phylogenetic analysis


> **_TASK:_** Execute the following command to submit the job to the cluster:

    sbatch job.sh

Then check its status with:

    squeue --me

Keep checking with that last command until your job is done, it will take a few minutes or longer. You should now have a series of additional files, all starting with the name of the sequence file (`siph16s.phy` in this case). A good overview of the primary `iqtree` output files is provided in the [iqtree tutorial](http://www.iqtree.org/doc/Tutorial).


## Download results

Now that you have done the computational heavy lifting on the cluster, you can move the files to your computer.

> **_Problem 1:_** Download the remote files to your local folder `exercise_04` that you will submit, and add this document to it. If you have all the provided files and output files from the above run in the submitted folder, you completed the problem.


## Inspect the phylogenetic tree

Of the files that `iqtree` generates, `siph16s.phy.treefile` is the actual phylogeny it inferred.

> **_Problem 2:_** Knit this rmd document to view the phylogeny with the code below. Note that this file needs to be in the same directory as the files you downloaded from the cluster for this code to work. If you produce knitted output with a tree, and the html output file is in this folder, you have completed this problem.

```{r fig.width=10, fig.height=14}

phy = read.tree("siph16s.phy.treefile")
plot(phy)

```

Note that we have just placed the name of the file in the `read.tree()` function. This approach differs from specifying the tree with `text=""` as we have in previous exercises, and is typical for important a tree from another program. The `plot()` function draws a phylogram.

> **_Problem 3:_** Answer the following questions about the phylogeny, and explain your reasoning in a sentence. Place your answer right below the question.

- Is the rate of evolution uniform for each species?

- Do the Agalma species form a monophyletic group? Why or why not?


## Inspect the report file

The report file, which has a name that ends with `.iqtree`, contains critical information about the phylogenetic inference analysis. The report file for this analysis is `siph16s.phy.iqtree`. Open this file in a text editor, such as VS Code.

There are a few sections of particular interest:

- `SEQUENCE ALIGNMENT` gives some statistics on the sequence alignment input file.

- `ModelFinder` shows the model evaluation results. Hundreds of models are evaluated, and the one that is found to have the best tradeoff between likelihood and the number of estimated parameters is indicated. We will explore the theory behind this process later in the course. The best model found for the data, which is used for the actual maximum likelihood search, is indicated on the line that starts:

    Best-fit model according to BIC: 

- `SUBSTITUTION PROCESS` gives the model parameter values that were estimated from the data.

- `MAXIMUM LIKELIHOOD TREE` section starts with the log-likelihood of the tree. It then presents the actual phylogeny, in a human readable ascii art format and then in newick format.

> **_Problem 4:_** Answer the following questions about the report file. Place your answer right below the question.

- What is the log-likelihood of your maximum likelihood tree?

- What is the substitution model selected by `iqtree`? There are multiple components to the model, each separated by a `+`. Consult the [iqtree DNA models page](http://www.iqtree.org/doc/Substitution-Models#dna-models) to read more about each component of your model. Explain what each component of your model is in sentence.



## Submit

Knit this file with all answers to your questions, compress this `exercise_04` folder (containing the original files you copied, `iqtree` output, and this file) with zip, and submit it for the Exercise 04 assignment.

